### PR TITLE
add optimal encoding to jpg files

### PR DESCRIPTION
**Pull request by pineapplespatula**
requires https://github.com/adobe-photoshop/generator-assets/pull/396

### CODE REVIEW

File: lib/convert.js [hunk 1]
```diff
@@ -167,7 +169,11 @@
         }
 
         // Write an image of format <format> to STDOUT
-        args.push(format + ":-");
+        if (format === 'jpg' && useFlite && settings.useOptimalJPEG) {
```
Comment (iwehrman) [inline lib/convert.js]: Drive-by formatting nit: you want double-quotes here, not single quotes.

File: lib/convert.js [hunk 2]
```diff
@@ -167,7 +169,11 @@
...
+            args.push(format + ":encode-optimal")
```
Comment (iwehrman) [inline lib/convert.js]: Travis is also complaining about a missing semicolon here.

File: lib/generator.js [hunk 3]
```diff
@@ -990,6 +991,7 @@
                 bounds:     true,
                 boundsOnly: settings.boundsOnly,
                 settings: settings.thread,
+                useOptimalJPEG: settings.useOptimalJPEG || true,
```
Comment (mcilroyc) [inline lib/generator.js]: doesn't this make it impossible to force it to `false`?

File: lib/convert.js [hunk 4]
```diff
@@ -167,7 +169,11 @@
         }
 
         // Write an image of format <format> to STDOUT
-        args.push(format + ":-");
+        if (format === "jpg" && useFlite && settings.useOptimalJPEG) {
```
Comment (mcilroyc) [inline lib/convert.js]: As I mentioned in the related PR, it seems like you should consider allowing a range of options (the range supported by FLITE) here instead of just toggling the "optimal" encoding.

**Comment by mcilroyc** [comment]
comments added, back to @pineapplespatula

File: lib/convert.js (hunk 4 above)
Comment (pineapplespatula) [inline lib/convert.js]: currently Save for web just enabled optimal or not, do you think we would want 

jpg:encode-fast or,
jpg:encode-precomputed

( I assume, perhaps incorrectly that the other option jpg:encode-default is the same as jpeg:- )

Comment (mcilroyc) [inline lib/convert.js]: Yes, I'm suggesting that we might want all three (four?) options.  For testing/comparing if nothing else.

Comment (pineapplespatula) [inline lib/convert.js]: apparently :...-fast is the same as :- and :...-default
I'll add precomputed tomorrow, although I am strongly against ever making it user facing ( which makes me wonder why to even add it)

**Comment by pineapplespatula** [comment]
back to @mcilroyc

File: lib/generator.js [hunk 5]
```diff
@@ -928,6 +928,7 @@
      * @param {!float}   settings.outputRect.top    Pixel distance of the rect's top from the doc's top
      * @param {!float}   settings.outputRect.right  Pixel distance of the rect's right side from the doc's left side
      * @param {!float}   settings.outputRect.bottom Pixel distance of the rect's bottom from the doc's top
+     * @param {?boolean} settings.useJPGEncoding Use aternamte huffman encoding, either optimal, or precomputed
```
Comment (mcilroyc) [inline lib/generator.js]: `@param {string=}`
(and again later)
